src/plugins/data/common/index_patterns/utils.ts

```diff
@@ -28,26 +28,78 @@
  * under the License.
  */
 
+import { DataSourceAttributes } from 'src/plugins/data_source/common/data_sources';
 import type { IndexPatternSavedObjectAttrs } from './index_patterns';
-import type { SavedObjectsClientCommon } from '../types';
+import type { SavedObject, SavedObjectReference, SavedObjectsClientCommon } from '../types';
 
 /**
  * Returns an object matching a given title
  *
  * @param client {SavedObjectsClientCommon}
  * @param title {string}
+ * @param dataSourceId {string}{optional}
  * @returns {Promise<SavedObject|undefined>}
  */
-export async function findByTitle(client: SavedObjectsClientCommon, title: string) {
+export async function findByTitle(
+  client: SavedObjectsClientCommon,
+  title: string,
+  dataSourceId?: string
+) {
```

Comment:
this is not correct query. The `searchQuery` will only be running against `searchFields`, which is `title` in this case. You wouldn't find `datasourceId` in the `title` field of an `index-pattern`, because datasource id exists in the `reference` field. I don't think by using simply query fit this use case